### PR TITLE
feat: support for attribute key suggestions and example queries in logs explorer query builder 

### PR DESCRIPTION
**Pull request by vikrantgupta25**
### Summary

- add support for attribute key ranked suggestions in the logs query builder search 
- support for example queries in the query builder ( logs explorer )
- added support for cancel current running query in the logs explorer page 

#### Related Issues / PR's

fixes - https://github.com/SigNoz/engineering-pod/issues/1597

#### Screenshots

https://github.com/user-attachments/assets/ffb805a8-cf22-4718-afdb-3b14fe91e082

https://github.com/user-attachments/assets/fe2a4a8d-3cf3-4606-9000-a5e6d93643a9

### CODE REVIEW

**Comment by SagarRajput-7** [comment]
Please add relevant images and video for reference

**Comment by GeekBoySupreme** [comment]
https://github.com/user-attachments/assets/8b51659e-769e-492e-a140-e69b8e55f95d
- There shouldn't be scrollable containers inside the dropdown. The list of suggestions we're showing need to be part of the dropdown's scroll. If that leads to example queries being lower in the list, that's an acceptable trade-off.
- Because of this 👆🏼 when I am using the keyboard down arrow, my selection never reaches the example queries.
- The attribute names don't need a tooltip here. 
- Please take a re-look at the spacing between the dots/icons and the attribute labels.
image
- This icon would be chevron down and when the dropdown is open, it can simply be chevron up. Does not need to change to magnifying glass.
image
- All icons used in the query builder are still antd icons. We need to migrate these to Lucide.

**Comment by vikrantgupta25** [comment]
@GeekBoySupreme addressed all the comments except one , starting discussion for the same on slack

File: frontend/src/api/queryBuilder/getAttributeSuggestions.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+				/=+$/,
+				'',
+			);
+		} catch {
+			// default base64 encoded string for empty filters object
+			base64EncodedFiltersString = 'eyJpdGVtcyI6W10sIm9wIjoiQU5EIn0';
+		}
+		const response: AxiosResponse<{
+			data: IGetAttributeSuggestionsSuccessResponse;
+		}> = await ApiV3Instance.get(
```
Comment (YounixM) [inline frontend/src/api/queryBuilder/getAttributeSuggestions.ts:37]: Do we want to support this only for v3?

Comment (vikrantgupta25) [reply frontend/src/api/queryBuilder/getAttributeSuggestions.ts:37]: yes this is only supported for v3 instance for now. cc - @raj-k-singh

File: frontend/src/constants/shortcuts/logsExplorerShortcuts.ts [hunk 2]
```diff
@@ -4,16 +4,19 @@ const userOS = getUserOperatingSystem();
...
 		userOS === UserOperatingSystem.MACOS ? 'cmd' : 'ctrl'
 	}+enter`,
 	FocusTheSearchBar: 's',
+	ShowAllFilters: `${userOS === UserOperatingSystem.MACOS ? 'cmd' : 'ctrl'}+/`,
 };
 
 export const LogsExplorerShortcutsDescription = {
 	StageAndRunQuery: 'Stage and Run the current query',
 	FocusTheSearchBar: 'Shift the focus to the last query filter bar',
+	ShowAllFilters: 'Toggle the all filters in the filters dropdown',
```
Comment (YounixM) [inline frontend/src/constants/shortcuts/logsExplorerShortcuts.ts:21]: `Toggle all filters in the filters dropdown`

File: frontend/src/container/QueryBuilder/components/ToolbarActions/ToolbarActions.styles.scss [hunk 3]
```diff
@@ -68,4 +112,17 @@
...
+		.loading-btn {
+			background: var(--bg-vanilla-300);
+		}
+
+		.cancel-run {
+			color: var(--bg-vanilla-100);
+		}
+
+		.cancel-run:hover {
+			background-color: #ff7875;
```
Comment (YounixM) [inline frontend/src/container/QueryBuilder/components/ToolbarActions/ToolbarActions.styles.scss:125]: Do we not have this color in the design tokens?

File: frontend/src/container/QueryBuilder/filters/QueryBuilderSearch/ExampleQueriesRendererForLogs.tsx [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+import { TagFilter } from 'types/api/queryBuilder/queryBuilderData';
+
+function ExampleQueriesRendererForLogs({
+	label,
+	value,
+	handleAddTag,
+}: ExampleQueriesRendererForLogsProps): JSX.Element {
+	return (
+		<div
+			className="container"
```
Comment (YounixM) [inline frontend/src/container/QueryBuilder/filters/QueryBuilderSearch/ExampleQueriesRendererForLogs.tsx:14]: use a specific class name

Comment (vikrantgupta25) [reply frontend/src/container/QueryBuilder/filters/QueryBuilderSearch/ExampleQueriesRendererForLogs.tsx:14]: done

File: frontend/src/container/QueryBuilder/components/ToolbarActions/ToolbarActions.styles.scss (hunk 3 above)
Comment (vikrantgupta25) [reply frontend/src/container/QueryBuilder/components/ToolbarActions/ToolbarActions.styles.scss:125]: no we do not have this token yet

**Review by YounixM — APPROVED** [review]
Minor comments - rest LGTM

File: frontend/src/container/QueryBuilder/components/ToolbarActions/ToolbarActions.styles.scss (hunk 3 above)
Comment (vikrantgupta25) [reply frontend/src/container/QueryBuilder/components/ToolbarActions/ToolbarActions.styles.scss:125]: @GeekBoySupreme are we supposed to use colors outside design tokens or are we adding this to the tokens ?

File: frontend/src/constants/shortcuts/logsExplorerShortcuts.ts (hunk 2 above)
Comment (vikrantgupta25) [reply frontend/src/constants/shortcuts/logsExplorerShortcuts.ts]: done

**Review by YounixM — APPROVED** [review]

**Pull request merged**